clowdapp.yaml

```diff
@@ -36,16 +36,18 @@ objects:
       enabled: true
       appName: "compliance"
       insightsOnly: true
-    deployments:
-    - name: db-migrations
+    jobs:
+    - name: compliance-import-ssg
+      schedule: "*/5 * * * *" # every 5 minutes
+      concurrencyPolicy: Forbid
       podSpec:
         image: ${IMAGE}:${IMAGE_TAG}
         initContainers:
         - command:
             - bash
             - -c
             - |
-              bundle exec rake --trace db:debug db:migrate
+              while ! bundle exec rake --trace db:abort_if_pending_migrations; do sleep 1; done;
```

Comment:
Can it wait up to some time out? The same as waits elsewhere with `for i in {1..120}; ...`?

Reply:
Sure, but why? What do you recommend doing on timeout? What's the problem with waiting infinitely? The timeout will only serve to exit(1) and cause openshift to restart the pod and retry the task. IMO that's more complex and more difficult to determine what's going on because we see multiple restarts just because it's still waiting.

Reply:
The restart is deliberate here to claim the service as failed once the timeout is reached. The reason for this is to fulfill an expectation that a migration would not take longer than expected. If it does take longer, we do not want to have it hang indefinitely, but rather fail it and restart its lifecycle. I feel like keeping it wait without this "timeout" could lead to some hanging problems. If for example the migration will take too long or would not finished (would be killed over and over), then OpenShift/Kubernetes after a several failures might back off for a while and release cluster's resources.

These 120 cycles might be enough for debugging. Anyhow, this is a busy wait that depends on some other pod/service.

Reply:
If the migrations fail, then the entire deployment will fail regardless of the state of this init container due to how clowder works. Your explanation does not convince me of any benefit; however, I'll make the change for consistency.

IMO we should either wait indefinitely (like this while loop) and not rely on openshift _or_ don't include any loop here and just let the pod crash until it's ready to start and rely completely on openshift. Doing some sort of hybrid complicates the deployment logic for no reason.